doc: Add 2 new FAQs (#2768)
projectacrn/acrn-hypervisor: doc/faq.rst

    hex "Size of the RAM region used by the hypervisor"
    default 0x07800000 if PLATFORM_SBL
    default 0x0f000000 if PLATFORM_UEFI

How to modify the default display output for a UOS?
******************************************************************************

> Apollo Lake HW has three pipes and each pipe can have three or four planes which helps to

Reviewer: ```suggestion
Apollo Lake HW has three pipes and each pipe can have three or four planes which help to
```

display the overlay video. The hardware can support up to 3 monitors simultaneously.

> Some parameters are available to control how display monitors are assigned between the SOS and UOS, simplifying the assignment policy and providing configuration flexibility for the pipes and planes in various IOT scenarios

Reviewer: For documentation, we prefer to keep the line length < 80 characters

* i915.avail_planes_per_pipe: for controlling the planes
* i915.domain_plane_owners: for controlling the domain mapping for each plane

> The detailed parameter description can refer to :ref:`GVT-g-kernel-options`

Reviewer: ```suggestion
Refer to :ref:`GVT-g-kernel-options` for detailed parameter descriptions.
```

Currently, pipe A is assigned to SOS and pipes B and C are assigned to UOS,
which uses the below parameter:

* SOS:

.. code-block:: bash

  i915.avail_planes_per_pipe=0x01010F
  i915.domain_plane_owners=0x011111110000

* UOS:

.. code-block:: bash

  i915.avail_planes_per_pipe=0x0070F00

If pipes A and B are assigned to UOS, while pipe C is assigned to SOS,
the below parameters can be used:

* SOS:

.. code-block:: bash

  i915.avail_planes_per_pipe=0x070101
  i915.domain_plane_owners=0x000011111111

* UOS:

.. code-block:: bash

  i915.avail_planes_per_pipe=0x000F0F

Why does ACRN need to be know how much RAM the system has?
************************************************************

> This is a tradeoff between flexibility and functionality certification. 

Reviewer: How about this rewrite:
```
Configuring ACRN at compile time with the system RAM size is
a tradeoff between flexibility and functionality certification. 
For server virtualization, one binary is typically used for all platforms
with flexible configuration options given at run time. 
But, for IoT applications, the image is typically configured and built
for a particular product platform and optimized product use. 

Important features for ACRN include functional safety (FuSa) and
real-time behavior. FuSa requires a static allocation policy to avoid
the potential of dynamic allocation failures.  Real-time applications
similarly benefit from static memory allocation. This is why ACRN
removed all "malloc" like code, and why it needs to pre-identify 
the size of all buffers and structures used in the Virtual Memory
Manager. For this reason, knowing the available RAM size at compile
time is necessary to statically allocate memory usage. 
```

    hex "Size of the RAM region used by the hypervisor"
    default 0x07800000 if PLATFORM_SBL
    default 0x0f000000 if PLATFORM_UEFI

> How to change the assigned output for one UOS instead of the default setting ?

Reviewer 2: ```suggestion
How to modify the default display output for a UOS
```

******************************************************************************

> BXT HW has three pipes and each pipe can have three or four planes which helps to

Reviewer 2: We never talk about `BXT` in our documentation, can we refer to `Apollo Lake` (or `APL`) instead?

It would also be great to have a document that describes the number of pipes/planes available. This is implicitly described in https://projectacrn.github.io/latest/developer-guides/GVT-g-kernel-options.html#i915-domain-plane-owners for Apollo Lake but I know this is not true for KBL, which is the other HW platform we support. Could we add such info (not necessarily in this PR).

display the overlay video. And it can support up to 3 monitors simultaneously.

> In order to assign the display monitors between SOS and UOS, some parameters are

Reviewer: @wenlingz Were you going to make this suggested change?

added to control the assign policy. It can help to simplify the assign policy

> while it provides the flexibility of configuring the pipe/planes for SOS/UOS under IOT scenario.

Reviewer 2: I think it would be worth mentionning too that this is known as the "plane restriction" feature. There are other scenarii possible to overlay content from different VMs across all display outputs.

* i915.avail_planes_per_pipe: uses to control the planes
* i915.domain_plane_owners: uses to control the domain mapping for each plane

> The detailed parameter description can refer to :ref:`GVT-g-porting`

Reviewer 2: While this is a great doc to provide background info, the one that really has the details is this:
```suggestion
The detailed parameter description can refer to :ref:`GVT-g-kernel-options`
```

Currently the pipe A is assigned to SOS and the pipe B/C is assigned to UOS,
which uses the below parameter:

* SOS:

::

  i915.avail_planes_per_pipe=0x01010F
  i915.domain_plane_owners=0x011111110000

* UOS:

::

  i915.avail_planes_per_pipe=0x0070F00

If the PIPE A/B is assigned to UOS while pipe  C is assigned to SOS,
the below parameters can be used:

* SOS:

::

  i915.avail_planes_per_pipe=0x070101
  i915.domain_plane_owners=0x000011111111

* UOS:

::

>   i915.avail_planes_per_pipe=0x000F0F

Reviewer 2: Could you add an example with 1x SOS and 2x UOS?

    hex "Size of the RAM region used by the hypervisor"
    default 0x07800000 if PLATFORM_SBL
    default 0x0f000000 if PLATFORM_UEFI

How to modify the default display output for a UOS?
******************************************************************************

Apollo Lake HW has three pipes and each pipe can have three or four planes which helps to
display the overlay video. And it can support up to 3 monitors simultaneously.
In order to assign the display monitors between SOS and UOS, some parameters are
added to control the assign policy(this is known as the "plane restriction" feature).
It can help to simplify the assign policy
while it provides the flexibility of configuring the pipe/planes for SOS/UOS under IOT scenario.

* i915.avail_planes_per_pipe: uses to control the planes
* i915.domain_plane_owners: uses to control the domain mapping for each plane

The detailed parameter description can refer to :ref:`GVT-g-kernel-options`

Currently the pipe A is assigned to SOS and the pipe B/C is assigned to UOS,
which uses the below parameter:

> * SOS:

Reviewer: @wenlingz Were you going to make this suggested change?  This way of showing the code blocks will render better than what you've written below.
